src/generate-social-cards.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * This file lops through every state and takes screenshots of the social card for each state
+ * This script is only meant to be run locally
+ */
+
+import { launch } from "puppeteer"
+import { existsSync, mkdirSync } from "fs"
+
+var waitingForReady = true
+
+const placeNames = [
+  "alabama",
+  "alaska",
+  "arizona",
+  "arkansas",
+  "california",
+  "colorado",
+  "connecticut",
+  "delaware",
+  "district_of_columbia",
+  "florida",
+  "georgia",
+  "hawaii",
+  "idaho",
+  "illinois",
+  "indiana",
+  "iowa",
+  "kansas",
+  "kentucky",
+  "louisiana",
+  "maine",
+  "maryland",
+  "massachusetts",
+  "michigan",
+  "minnesota",
+  "mississippi",
+  "missouri",
+  "montana",
+  "nebraska",
+  "nevada",
+  "new_hampshire",
+  "new_jersey",
+  "new_mexico",
+  "new_york",
+  "north_carolina",
+  "north_dakota",
+  "ohio",
+  "oklahoma",
+  "oregon",
+  "pennsylvania",
+  "rhode_island",
+  "south_carolina",
+  "south_dakota",
+  "tennessee",
+  "texas",
+  "utah",
+  "vermont",
+  "virginia",
+  "washington",
+  "west_virginia",
+  "wisconsin",
+  "wyoming",
+]
+
+/*
+ * For each state, use puppeteer to capture a screnshot
+ * @param name {strign} the name of the state to have the social-card taken a picture of
+ */
+async function captureScreenshot (name) {
+  let browser = null
+  waitingForReady = false
+  let outputDir = "static/social-cards"
+  // if screenshots directory is not exist then create one
+  if (!existsSync(outputDir)) {
+    mkdirSync(outputDir)
+  }
+  try {
+    // launch headless Chromium browser
+    browser = await launch({ headless: false })
+    // create new page object
+    const page = await browser.newPage()
+    // set viewport width and height
+    await page.setViewport({ width: 800, height: 418 })
+
+    //no timeout time in case it takes a page a long time to load
+    page.setDefaultNavigationTimeout(0)
+
+    //direct puppeteer to the social-card page for each state
+    await page.goto("http://localhost:8000/social-card?state=" + name, {
+      waitUntil: "networkidle2",
+    })
+
+    // capture screenshot and store it into static/social-cards directory.
+    await page.screenshot({ path: outputDir + "/" + name + ".jpg" })
+    await browser.close()
+    waitingForReady = true
+  } catch (err) {
+    console.log(`❌Error: ${err.message}`)
+  }
+}
+
+//wait for the previous state to finish taking screenshot before making the next one
+setTimeout(async function () {
+  for (const name of placeNames) {
+    while (!waitingForReady) {
+      setTimeout(function () {}, 1000)
+    }
+    await captureScreenshot(name)
+  }
+}, 3000)
```

Comment:
There's an idea in code about avoiding magic strings and magic numbers, which are numbers in code that are uncelar. It's best to move this to a constant, like `const SocialPageDelayMs`. Similarly for the `1000` you can call that something like `WaitDelayMs`.